### PR TITLE
SDMEXT-560:Skeleton for CAP Java Plugin

### PR DESCRIPTION
**Pull request by rashmiangadi11**
SDMEXT-560:Basic skeleton as maven project with test,resources and java folders

### CODE REVIEW

**Comment by dheerasameerk** [comment]
@rashmiangadi11 

1. What is the reason of creating java package as com.sap.cds.sdm?
2. Can the integration tests folder reside under src/main/test or is it required to be in the root?

File: integration-tests/.gitignore
Comment (yashmeet29) [inline integration-tests/.gitignore:1]: Shouldn't this file be on root level?

**Review by dheerasameerk — APPROVED** [review]

File: sdm/src/main/resources/cds/com.sap.cds/sdm/attachments.cds [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace sap.attachments;
+
+using {
+    cuid,
+    managed
+} from '@sap/cds/common';
+
+
+aspect MediaData           @(_is_media_data) {
+    content   : LargeBinary; // stored only for db-based services
```
Comment (maheshsooryambylu) [inline sdm/src/main/resources/cds/com.sap.cds/sdm/attachments.cds:10]: Do we need to store the large binary ? We are only transferring the content via stream operation isnt it ?

**Review by maheshsooryambylu — COMMENTED** [review]
Would like to understand  why attachment need to be designed as a Large binary with the intent of storing instead of being just a stream operation between the client and the DI service ? (to better understand from performance optimization perspective)

File: sdm/src/main/resources/cds/com.sap.cds/sdm/attachments-annotations.cds [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        {Value: createdAt},
+        {Value: createdBy},
+        {Value: note}
+    ]
+} {
+    note       @(title: '{i18n>attachment_note}');
+    modifiedAt @(odata.etag);
+}
+
+annotate Attachments with @Common: {SideEffects #ContentChanged: {
```
Comment (maheshsooryambylu) [inline sdm/src/main/resources/cds/com.sap.cds/sdm/attachments-annotations.cds:39]: What is the significance of this annotation ?

File: sdm/src/main/resources/cds/com.sap.cds/sdm/attachments.cds (hunk 1 above)
Comment (rashmiangadi11) [reply sdm/src/main/resources/cds/com.sap.cds/sdm/attachments.cds:10]: As discussed if we are storing in drafts for sometime this is required

File: sdm/src/main/resources/cds/com.sap.cds/sdm/attachments-annotations.cds (hunk 2 above)
Comment (rashmiangadi11) [reply sdm/src/main/resources/cds/com.sap.cds/sdm/attachments-annotations.cds:39]: I will check this once my POC is done

**Review by maheshsooryambylu — APPROVED** [review]

**Review by yashmeet29 — APPROVED** [review]

**Pull request merged**